.github/workflows/test.yml

```diff
@@ -2,30 +2,46 @@ name: CI
 
 on: [push, pull_request]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental }}
     strategy:
       fail-fast: false
       matrix:
-        emacs_version:
-          - 25.3
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        emacs-version:
           - 26.3
           - 27.2
           - 28.2
           - 29.3
-          - snapshot
```

Comment:
@jcs090218 why are we no longer testing snapshot?

Reply:
The snapshot is being tested, see L22 - L29. 

I've omitted the snapshot test on Windows since there is an upstream bug; see https://github.com/emacs-eask/cli/issues/224.